### DISCUSSION TITLE
'docker compose run archivebox [subcommand] [-args]' results orphan containers

### DISCUSSION THREAD

**Discussion by kouryoubon** [ArchiveBox/ArchiveBox#1623] [Q&A] [answered]
I just started using archivebox and I am also new to docker. one thing I notice is that since every time I run `docker compose run archivebox ...`command, docker will create a container and after the command finishes the container is exited. Now if I run a new `docker compose run archivebox ...` command it will prompt an orphan container warning.

Now I do know it can be resolved either by adding `--remove-orphans` flag or adding `-rm` flag to the command or running `docker container prune`. I also notice that the tutorials or issues or discussion did not mention this phenomenon.

So my question is does this would hurt performance? since every command I run will start a container and then the container is exited, waiting to be manually pruned, and what are the design choices and thoughts the behind this?

**Comment by stayupthetree** [comment, marked as answer]
Its a standard behavior of docker compose run and nothing to do with ArchiveBox. You could alternatively console into a running archivebox container and run the archivebox command, or just run the docker compose run command with the mentiioned --remove-orphans each time.

**Reply by kouryoubon** [reply]
I see, thanks for your reply

**Reply by pirate** [reply]
Or what I usually do is just let them acumulate and run `docker system prune --all` every now and then to clear dead containers across the entire system. I don't add `--rm` to all the example commands in the docs because it would make every example command longer and usually people that care about it know to add that flag themselves.
